src/main/java/ca/uhn/fhir/jpa/starter/CollectDataResourceProvider.java

```diff
@@ -0,0 +1,155 @@
+
+        parameters.addParameter(new Parameters.ParametersParameterComponent()
+                .setName(PlirConstants.PARAMETER_NAME_RESOURCE).setResource(observationBundle));
+
+        return parameters;
+    }
+
+    public MeasureReport createMeasureReport(String url) throws ClientProtocolException, IOException {
+
+        MeasureReport report = new MeasureReport();
+        report.setStatus(MeasureReport.MeasureReportStatus.COMPLETE);
+        report.setType(MeasureReport.MeasureReportType.DATACOLLECTION);
+        report.setMeasure(PlirConstants.MEASURE_RESOURCE_MEASURE);
+
+        Bundle bundle = fetchBundle(url);
+
+        List<Reference> references = new ArrayList<Reference>();
+
+        for (Bundle.BundleEntryComponent entry : bundle.getEntry()) {
+            if (entry.getResource() instanceof Observation) {
+                Observation resource = (Observation) entry.getResource();
+                Reference subjectRef = resource.getSubject();
+                Reference encounterRef = resource.getEncounter();
+                references.add(subjectRef);
+                references.add(encounterRef);
+            }
+        }
+        report.setEvaluatedResource(references);
+        return report;
+    }
+
+    public Bundle createObsBundle(String url) throws ClientProtocolException, IOException {
+        Bundle observationBundle = new Bundle();
+        Bundle bundle = fetchBundle(url);
+        observationBundle.setType(Bundle.BundleType.COLLECTION);
+
+        for (Bundle.BundleEntryComponent entry : bundle.getEntry()) {
+            if (entry.getResource() instanceof Observation) {
+                Resource resource = entry.getResource();
+                Bundle.BundleEntryComponent component = observationBundle.addEntry();
+                component.setResource(resource);
+            }
+        }
+        return observationBundle;
+    }
+
+    private String constructUrl(HttpServletRequest theServletRequest, Measure measure, String periodStart,
+            String periodEnd) {
+        StringBuffer baseUrl = new StringBuffer(theServletRequest.getScheme());
+        baseUrl.append("://");
+        baseUrl.append(theServletRequest.getServerName());
+        baseUrl.append(":");
+        baseUrl.append(theServletRequest.getServerPort());
+
+        String expression = measure.getGroupFirstRep().getPopulationFirstRep().getCriteria().getExpression();
+
+        StringBuffer params = new StringBuffer("&date=ge");
+        params.append(periodStart);
+        params.append("&date=lt");
+        params.append(periodEnd);
+
+        String fullUrl = baseUrl + "/fhir/" + expression + params;
+
+        return fullUrl;
+    }
+
+    private Bundle fetchBundle(String theUrl) throws IOException, ClientProtocolException {
+        Bundle bundle;
+        ourHttpClient = HttpClientBuilder.create().build();
+        HttpGet get = new HttpGet(theUrl);
+
+        String username = "hapi";
+        String password = "hapi123";
```

Comment:
these should not be hard coded

Reply:
@k-joseph ideally for now we use [hard coded ](https://github.com/openmrs/openmrs-contrib-plm-fhir-server/blob/master/src/main/java/ca/uhn/fhir/jpa/starter/AuthenticationInterceptor.java#L47  )credentials for Basic Auth just for the POC

we shall probably implement a better Authenication mechanism as a separate Task

Reply:
am ok with basic auth, but u need to details out of the code, get this into a properties file forexample

Reply:
sure makes sense , let me create a ticket for that fix. 
@gitcliff    will work on it

Reply:
and specifically here , i no longer read the hard coded creds, i simply pass the credentials directly from the Incoming request to the Client  . But we need to catch that here  https://github.com/openmrs/openmrs-contrib-plm-fhir-server/blob/master/src/main/java/ca/uhn/fhir/jpa/starter/AuthenticationInterceptor.java#L47  .

Created issue for Cliff https://issues.openmrs.org/browse/FM2-339